handlers/oauth/oauth_test.go

```diff
@@ -0,0 +1,286 @@
+package oauth
+
+import (
+	"encoding/json"
+	"errors"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"os"
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/gin-gonic/gin"
+	"github.com/markbates/goth"
+	"github.com/speedrun-website/leaderboard-backend/model"
+)
+
+type mockOauthStore struct {
+	Users map[uint]*model.User
+}
+
+func makePointerString(str string) *string {
+	return &str
+}
+
+var rollingID uint = 0
+
+func initStore() *mockOauthStore {
+	usersMap := make(map[uint]*model.User)
+	usersMap[rollingID] = &model.User{
+		Username:  "FastMcGo",
+		Email:     "fast@example.com",
+		TwitterID: makePointerString(strconv.Itoa(int(rollingID))),
+	}
+	rollingID++
+
+	store := mockOauthStore{
+		Users: usersMap,
+	}
+	Oauth = store
+	return &store
+}
+
+func (s mockOauthStore) GetUserByTwitterID(twitterID string) (*model.User, error) {
+	if twitterID == "error" {
+		return nil, errors.New("Issue finding user")
+	}
+
+	for _, user := range s.Users {
+		if *user.TwitterID == twitterID {
+			return user, nil
+		}
+	}
+	return nil, nil
+}
+
+func (s mockOauthStore) CreateUser(user model.User) (*model.User, error) {
+	if strings.Contains(user.Email, "error") {
+		return nil, errors.New("Issue creating user")
+	}
+	rollingID++
+	newUser := &model.User{
+		ID:        rollingID,
+		Email:     user.Email,
+		Username:  user.Username,
+		TwitterID: makePointerString(strconv.Itoa(int(rollingID))),
+	}
+	s.Users[rollingID] = newUser
+	return newUser, nil
+}
+
+func Test_OauthCallbackUserAuthError(t *testing.T) {
```

Comment:
I'd prefer the test function names not to have underscores, and I think that's it's suggested to just name things as `TestName` (see https://golang.org/doc/tutorial/add-a-test)

Reply:
Got rid of them underscores in https://github.com/leaderboardsgg/leaderboard-backend/pull/69/commits/44de6ebe33d546c4f8ac172dc6bcae6c25bd6e7a